src/meta/src/rpc/server.rs

```diff
@@ -171,308 +158,141 @@ pub async fn rpc_serve_with_store<S: MetaStore>(
                 leader_addr.host,
                 leader_addr.port
             );
-
-            if note_status_leader_rx.changed().await.is_err() {
-                panic!("Issue receiving leader value from channel");
-            }
         }
     });
 
-    // FIXME: Start leader services if follower becomes leader
-    // failover logic
+    // FIXME: maybe do not use a channel here
+    // What I need is basically a oneshot channel with one producer and multiple consumers
+    let (svc_shutdown_tx, mut svc_shutdown_rx) = WatchChannel(());
 
-    // Start follower services
-    if !leader_rx.borrow().1 {
-        tracing::info!("Node initially elected as follower");
-        tokio::spawn(async move {
-            let span = tracing::span!(tracing::Level::INFO, "services");
-            let _enter = span.enter();
+    let join_handle = tokio::spawn(async move {
```

Comment:
Can we move the follow service to a separate file like `leader_svs` to make it more clear?

Reply:
Sure. I created `follower_svc.rs`. This function there also directly starts the services instead of returning a `Router`. Let me know if you agree with that approach.